exploration/selection-declaration.md

```diff
@@ -149,14 +150,11 @@ _What prior decisions and existing conditions limit the possible design?_
 
 ## Proposed Design
 
-_Describe the proposed solution. Consider syntax, formatting, errors, registry, tooling, interchange._
+The design alternative [Match on variables instead of expressions](#match-on-variables-instead-of-expressions)
+described below is selected.
```

Comment:
I don't think I agree with this design alternative, since it *requires* the use of declarations.

Reply:
The way I see it, that's the least worst option that we have available. :/

Reply:
Okay, but it's a breaking change. The other alternatives have quirks for sure (and I think the mutable/immutable labels might be backwards??), but we need to seek consensus.

Reply:
I agree. On this topic, it seems like we've been avoiding expressing any preferences one way or another for a while, so I didn't know of a better way to start seeking that consensus than proposing one solution above the others. The PR is structured in a way that it should be relatively easy for us to change the selection from my initial proposition, should our consensus land elsewhere.

Reply:
On the point of formatting for selection != formatting for display, I dug up this ["virtual whiteboard" doc](https://docs.google.com/document/d/1nF1mNR1ahNCrsAe_ENfBieT1syWrTXw95MqQ4V-WIxQ/edit) from older times of MFWG when we couldn't meet in person (July 2020, to be exact), in which @mihnita and I had a kind of Socratic dialogue where Mihai explained to me through examples why selectors and placeholders are distinct operations and thus cannot be collapsed into a single construct type in the data model.